internal/ingress/metric/collector/collector.go

```diff
@@ -273,6 +283,49 @@ func (sc *SocketCollector) Run() {
 	}
 }
 
+// RemoveMetrics deletes prometheus metrics from prometheus for endpoints that
+// are not available anymore.
+// Ref: https://godoc.org/github.com/prometheus/client_golang/prometheus#CounterVec.Delete
+func (sc *SocketCollector) RemoveMetrics(endpoints []string) {
+	mfs, err := prometheus.DefaultGatherer.Gather()
+	if err != nil {
+		glog.Errorf("error gathering metrics:", err)
```

Comment:
(nit) It's clearer to start log messages with a capital letter to differentiate them from returned errors.

Reply:
done